product_kosher/models/product_kosher.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+from openerp.exceptions import Warning as UserError
+
+
+class ProductKosher(models.Model):
+    _name = "product.kosher"
+    _description = "Product Kosher"
+    _order = "date_start desc"
+
+    name = fields.Char(
+        string="# Certificate",
+        required=True,
+    )
+    issuer_id = fields.Many2one(
+        string="Issuer",
+        comodel_name="res.partner",
+        )
+    product_tmpl_id = fields.Many2one(
+        string="Product Template",
+        comodel_name="product.template",
+        ondelete="cascade",
+        required=True
+    )
+    date_start = fields.Date(
+        string="Start Date",
+        required=True
+    )
+    date_end = fields.Date(
+        string="End Date"
+    )
+
+    _sql_constraints = [
+        ("date_unique",
+         "unique(product_tmpl_id,date_start)",
+         _("No duplicate date start"),
+         ),
+    ]
+
+    @api.constrains(
+        "date_start", "date_end")
+    def _check_date(self):
+        strWarning = _(
```

Comment:
No reason to assign this to a var - you can just directly use in the UserError. Otherwise you're running the translation method in all instances of this method being run, but only using the output sometimes.